docs/rules/unbound-method.md

```diff
@@ -0,0 +1,53 @@
+# Enforces unbound methods are called with their expected scope (`unbound-method`)
```

Comment:
this needs to mention that the user needs to depend on `@typescript-eslint/eslint-plugin`